### PR TITLE
Add tests for Transaction class

### PR DESCRIPTION
**Pull request by ajwalkiewicz**
## Summary by Sourcery

Add comprehensive unit tests for the Transaction class and make it immutable by setting frozen=True in the dataclass decorator.

Enhancements:
- Make the Transaction class immutable by adding the frozen=True parameter to the dataclass decorator.

Tests:
- Add comprehensive unit tests for the Transaction class, covering initialization, equality, invalid inputs, and edge cases.

### CODE REVIEW

File: tests/test_transaction.py [hunk 1]
```diff
@@ -7,6 +7,82 @@
...
+class TestTransaction(unittest.TestCase):
+
+    def setUp(self):
+        """Set up a valid Transaction instance for testing."""
+        self.transaction1 = Transaction(date='2023-10-01T12:00:00', amount=100.50, description='Test Transaction 1')
+        self.transaction2 = Transaction(date='2023-10-02T12:00:00', amount=100.50, description='Test Transaction 2')
+        self.transaction3 = Transaction(date='2023-10-01T12:00:00', amount=200.00, description='Test Transaction 3')
+
+    def test_transaction_initialization(self):
+        """Test if Transaction is initialized correctly."""
```
[the remark this answers is not in the document]
Comment (ajwalkiewicz) [reply tests/test_transaction.py:20]: Transaction is designed in a way, that each instance has unique hash, based on its id. Thus 2 equal Transactions does not have same hash.
